### PR TITLE
[KYUUBI #4738] `AdminResource.getEngineSpace` should use primary group name on GROUP share level

### LINKED ISSUE #4738: [Bug] Cannot get the `GROUP` shared level Engine info using admin RESTful API.
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Details are here: https://github.com/apache/kyuubi/discussions/4684

### Affects Version(s)

1.7.0 / master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by HaoYang670**
### _Why are the changes needed?_

Closes #4738

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by HaoYang670** [comment]
I will add `test("list engine - group share level")` in `AdminResourceSuite`

File: .gitignore [hunk 1]
```diff
@@ -32,6 +32,8 @@
 .ensime_lucene
 .generated-mima*
 .vscode/
+.bloop/
```
Comment (zwangsheng) [inline .gitignore:35]: Please not introduce extraneous changes

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala [hunk 2]
```diff
@@ -355,9 +355,15 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
 
   private def getEngineSpace(engine: Engine): String = {
     val serverSpace = fe.getConf.get(HA_NAMESPACE)
+    val owner = if (engine.getSharelevel == "GROUP") {
```
Comment (zwangsheng) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:358]: Reference `EngineRef`, using the scala syntax
```
engine.getSharelevel match {
    case XXX =>
       // do something
    case  _ =>
       // do something
}
```

File: .gitignore (hunk 1 above)
Comment (pan3793) [reply .gitignore:35]: what's this for? vscode?

Comment (HaoYang670) [reply .gitignore:35]: This is for scala metals: https://github.com/scalameta/metals
I will remove it if you think it is not good to add them in this PR

Comment (pan3793) [reply .gitignore:35]: it's fine, but better in a separated PR

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala (hunk 2 above)
Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala:358]: the SERVER mode is not correct either

File: .gitignore (hunk 1 above)
Comment (HaoYang670) [reply .gitignore]: Cleaned

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (HaoYang670) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala]: Updated. Thank you!

Comment (HaoYang670) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala]: > the SERVER mode is not correct either

Could this be in another PR?

**Review by zwangsheng — COMMENTED** [review]
LGTM, would you like to add a unit test of the list engine and delete engine in Group level?

You can refer to `kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala`

**Comment by zwangsheng** [comment]
Also, cc @lightning-L as original author

**Review by pan3793 — APPROVED** [review]

**Comment by HaoYang670** [comment]
> would you like to add a unit test of the list engine and delete engine in Group level?

Sure I'd like to. But could you please help on how to implement a `GroupProvider` in the unit test?
I've found you are using the "hadoop" provider, how should I reuse it in the group share level?

**Comment by zwangsheng** [comment]
> > would you like to add a unit test of the list engine and delete engine in Group level?
> 
> Sure I'd like to. But could you please help on how to implement a `GroupProvider` in the unit test? I've found you are using the "hadoop" provider, how should I reuse it in the group share level?
> 
> https://github.com/apache/kyuubi/blob/6ba8b99151adeee2fbdc7127a1ec1b0a087d4787/kyuubi-server/src/test/scala/org/apache/kyuubi/server/api/v1/AdminResourceSuite.scala#L266-L269

According to current impl, you can also get `fe` in suite (You may need cast `fe` => `KyuubiRestFrontendService`).

Then `fe.sessionManager.groupProvider`

**Comment by HaoYang670** [comment]
How can I get the name of the primary group of the `Utils.currentUser`?

**Comment by pan3793** [comment]
the code LGTM, please run `dev/reformat` to fix the style

**Comment by HaoYang670** [comment]
fixed

**Review by zwangsheng — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Comment by HaoYang670** [comment]
Hi @bowenliang123. I create #4770 to track this. But I am afraid I can't give a quick fix because it is a low priority in my task list. Sorry from that.

**Comment by huangzhir** [comment]
> Hi @bowenliang123. I create #4770 to track this. But I am afraid I can't give a quick fix because it is a low priority in my task list. Sorry from that.

@HaoYang670 @bowenliang123 I have some free time, can I fix this bug https://github.com/apache/kyuubi/issues/4770 ?

**Comment by bowenliang123** [comment]
> > Hi @bowenliang123. I create #4770 to track this. But I am afraid I can't give a quick fix because it is a low priority in my task list. Sorry from that.
> 
> @HaoYang670 @bowenliang123 I have some free time, can I fix this bug #4770 ?

Cool, thanks for giving a hand in this. If we won't be able to fix this soon, I think we shall consider reverting this on branch-1.7 to prevent blocking the release process.

**Pull request closed without merging**